src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -162,26 +161,38 @@ protected void buildEndpoint(ApiEndpoint baseEndpoint) {
 
         Auth0UserProfile requestingUser = getUserFromRequest(req);
         if (isUserAdmin(requestingUser)) {
+            // If the user is admin, the context is presumed to be the admin dashboard, so we deliver all entities for
+            // management or review without restriction.
             return persistence.getAll();
-        } else if (persistence.clazz == OtpUser.class) { 
+        } else if (persistence.clazz == OtpUser.class) {
+            // If the required entity is of type 'OtpUser' the assumption is that a call is being made via the
+            // OtpUserController. Therefore, the request should be limited to return just the entity matching the
+            // requesting user.
             Bson filter = Filters.eq("_id", requestingUser.otpUser.id);
```

Comment:
The filter should be on the field `userId`, right? Perhaps this should be factored into a method (it's used below and there may be other occurrences as well.

Reply:
@landonreed  - "_id" is correct. The filter works on the columns used in Mongo. For an OTP user, the userId is held under the _id column. I have added a method to return the objects based on the these filter requirements.

Reply:
Right, ok. I think my confusion was around the fact that we're requesting "many" but filtering on the "_id" field, which by design should only return one result.